lua/telescope/utils.lua

```diff
@@ -149,7 +218,7 @@ utils.diagnostics_to_tbl = function(opts)
   local buffer_diags = opts.get_all and vim.lsp.diagnostic.get_all() or
     {[current_buf] = vim.lsp.diagnostic.get(current_buf, opts.client_id)}
   for bufnr, diags in pairs(buffer_diags) do
-    for _, diag in pairs(diags) do
+    for _, diag in ipairs(diags) do
```

Comment:
I actually think this occasion should be `pairs`.

Short explanation:
`ipairs` does this:
- start from idx 1 til tbl[idx] == nil
- if `buffer_diags now: `{[current_buf] = vim.lsp.diagnostic.get(current_buf, opts.client_id)}` and `current_buf ~= 1` `ipairs` does nothing.
- `pairs` will iterate over all elements in the table, just not sorted.
- So `ipairs` if it is a list starting at 1 ... n else `pairs`